packages/core/src/InlineEditor/InlineEditor.tsx

```diff
@@ -128,15 +139,75 @@ export const HvInlineEditor = fixedForwardRef(function HvInlineEditor<
       newValue = cachedValue;
       setEditMode(false);
       setValue(newValue);
+      checkOverflow();
     }
     onKeyDown?.(event, newValue);
   };
 
   const handleChange: HvInputProps["onChange"] = (event, val) => {
     setValue(val);
+    checkOverflow();
     onChange?.(event, val);
   };
 
+  const ButtonComponent = useMemo(() => {
+    return (
+      <HvButton
+        ref={btnRef}
+        variant="secondaryGhost"
+        overrideIconColors={false}
+        endIcon={
+          <Edit
+            color="secondary_60"
+            className={cx(classes.icon, {
+              [classes.iconVisible]: showIcon,
+            })}
+          />
+        }
+        className={cx(classes.button, {
+          [classes.largeText]: parseInt(lineHeight as string, 10) >= 28,
+        })}
+        onClick={handleClick}
+        disabled={disabled}
+        {...buttonProps}
+      >
+        <HvTypography
+          variant={variant}
+          noWrap
+          className={cx(classes.text, { [classes.textEmpty]: !value })}
+          {...typographyProps}
+        >
+          {value || placeholder}
+        </HvTypography>
+      </HvButton>
+    );
+  }, [
+    buttonProps,
+    classes.button,
+    classes.icon,
+    classes.iconVisible,
+    classes.largeText,
+    classes.text,
+    classes.textEmpty,
+    cx,
+    disabled,
+    handleClick,
+    lineHeight,
+    placeholder,
+    showIcon,
+    typographyProps,
+    value,
+    variant,
+  ]);
+
+  const renderButton = () => {
+    return isOverflowing ? (
+      <HvTooltip title={value}>{ButtonComponent}</HvTooltip>
```

Comment:
we can pass a falsy value to not render the tooltip
```suggestion
    return (
      <HvTooltip title={isOverflowing && value}>{ButtonComponent}</HvTooltip>
```